bugsnag/src/main/java/com/bugsnag/Report.java

```diff
@@ -131,23 +131,22 @@ public String getContext() {
 
     @Expose
     Map<String, Object> getSession() {
+        return sessionMap;
+    }
+
+    void setSession(Session session) {
         if (session == null) {
-            return null;
+            sessionMap = null;
```

Comment:
We should add a unit test to verify that if the session parameter is null, the payload doesn't contain any session information. The current implementation will cause a crash due to NPEs.